src/block/context.rs

```diff
@@ -0,0 +1,650 @@
+//! The Context module holds all the functionality for the test declaration, that is:
+//! `before`, `after`, `suite`, `context`, `it` and their variants.
+//!
+//! A Context can also holds reference to children Contextes, for whom the before closures will be
+//! executed after the before closures of the current context. The order of execution of tests
+//! respect the order of declaration of theses tests.
+//!
+//! Running these tests and doing asserts is not the job of the Context, but the Runner.
+//!
```

Comment:
👍

Reply:
This should mention that for parallel execution the order is undefined, for obvious reasons. Consider it fixed.